railties/lib/rails/configuration.rb

```diff
@@ -64,9 +65,10 @@ def unshift(*args, &block)
       end
 
       def merge_into(other) #:nodoc:
-        @operations.each do |operation, args, block|
+        @operations.concat(@delete_operations).each do |operation, args, block|
```

Comment:
I think to address the concerns we could just either create a new array and just loop through both typo of operations.